modules/boulder/src/main/ts/ephox/boulder/core/ValueProcessor.ts

```diff
@@ -1,262 +1,203 @@
-import { Adt, Arr, Fun, Merger, Obj, Optional, Thunk, Type } from '@ephox/katamari';
+import { Arr, Fun, Merger, Obj, Optional, Thunk, Type } from '@ephox/katamari';
 import { SimpleResult, SimpleResultType } from '../alien/SimpleResult';
-
 import * as FieldPresence from '../api/FieldPresence';
 import * as Objects from '../api/Objects';
 import { ResultCombine } from '../combine/ResultCombine';
-import * as ObjWriter from './ObjWriter';
 import * as SchemaError from './SchemaError';
+import * as ValuePresence from './ValuePresence';
 
 type SchemaError = SchemaError.SchemaError;
 
-// TODO: Handle the fact that strength shouldn't be pushed outside this project.
-export type Strength = (res: any) => any;
-export type ValueValidator = (a, strength?: Strength) => SimpleResult<string, any>;
-export type PropExtractor = (path: string[], strength: Strength, val: any) => SimpleResult<SchemaError[], any>;
-export type ValueExtractor = (label: string, prop: Processor, strength: Strength, obj: any) => SimpleResult<SchemaError[], string>;
+export type ValueValidator = (a) => SimpleResult<string, any>;
+export type PropExtractor = (path: string[], val: any) => SimpleResult<SchemaError[], any>;
+export type ValueExtractor = (label: string, prop: Processor, obj: any) => SimpleResult<SchemaError[], string>;
 export interface Processor {
-  extract: PropExtractor;
+  extractProp: PropExtractor;
   toString: () => string;
 }
 
-export type FieldValueProcessor<T> = (key: string, okey: string, presence: FieldPresence.FieldPresenceAdt, prop: Processor) => T;
-export type StateValueProcessor<T> = (okey: string, instantiator: (obj: any) => Optional<unknown>) => T;
-
-export interface ValueProcessorAdt {
-  fold: <T>(
-    field: FieldValueProcessor<T>,
-    state: StateValueProcessor<T>
-  ) => T;
-  match: <T>(branches: {
-    field: FieldValueProcessor<T>;
-    state: StateValueProcessor<T>;
-  }) => T;
-  log: (label: string) => void;
-}
-export type FieldProcessorAdt = ValueProcessorAdt;
```

Comment:
Note this alias existed (because of legacy things where both were maybe separate or something) and I removed it because unnecessary and confusing. The aliased type name is the one that was used outside of boulder.